src/clj/muuntaja/core.clj

```diff
@@ -82,8 +82,9 @@
                 ;; optional guard on empty imput
                 (if (and allow-empty-input-on-decode? decode?)
                   (fn [^InputStream is charset]
-                    (if (pos? (.available is)) ($ is charset)))
-                  $))
+                    (if (and (not (nil? is)) (pos? (.available is))) ($ is charset)))
+                  (fn [^InputStream is charset]
+                    (if-not (nil? is) ($ is charset)))))
```

Comment:
Ah, it's not just for the http-stuff. It also effects normal standalone decode. It doesn't have request, just data.

```clj
(m/decode (m/create) "application/json" nil)
; nil
```
So, it should be called....?